packages/SwingSet/test/test-vpid-kernel.js

```diff
@@ -301,25 +334,28 @@ async function doTest123(t, which, mode) {
   // before resolution, A's c-list should have the promise
   t.equal(inCList(kernel, vatA, p1kernel, p1VatA), true);
 
-  doResolveSyscall(syscallA, p1VatA, mode, rootBvatA);
+  const targetsA = {
+    target2: rootBvatA,
+    localTarget: localTargetA,
+    p1: dataPromiseA,
+  };
+  doResolveSyscall(syscallA, p1VatA, mode, targetsA);
   await kernel.run();
 
-  t.deepEqual(logB.shift(), resolutionOf(p1VatB, mode, rootBvatB));
+  const targetsB = {
+    target2: rootBvatB,
+    localTarget: localTargetB,
+    p1: dataPromiseB,
+  };
+  t.deepEqual(logB.shift(), resolutionOf(p1VatB, mode, targetsB));
   t.deepEqual(logB, []);
 
-  // TODO: once kernel->vat resolution notification retires clist entries,
-  // switch to the other set of assertions
-
-  // after resolution, (before we implement retirement), A's c-list should
-  // still have the promise
-  t.equal(inCList(kernel, vatA, p1kernel, p1VatA), true);
-
-  // after resolution, (now that we've implemented retirement), A's c-list
-  // should *not* have the promise
-  // t.equal(inCList(kernel, vatA, p1kernel, p1VatA), false);
-  // t.equal(clistKernelToVat(kernel, vatA, p1kernel), undefined);
-  // t.equal(clistVatToKernel(kernel, vatA, p1VatA), undefined);
-
+  if (expectRetirement) {
+    // after resolution, A's c-list should *not* have the promise
+    t.equal(inCList(kernel, vatA, p1kernel, p1VatA), false);
```

Comment:
No, I think that latter bit is wrong.  We really do want to look at the specific clist entry, because a promise my be retired in one vat but not in another, and so we would expect the entry to be removed from the one vat's clist and not the others.

Reply:
On second thought, looking at the code it's checking both inCList AND clistKernelToVat/clistVatToKernel, so the expectations of universality were already there in the case of this test.